### DISCUSSION TITLE
Unified Axis Keywords syntax

### DISCUSSION THREAD

**Discussion by VorTechnix** [sbrl/Minetest-WorldEditAdditions#79] [Ideas]
## Axis handling:
For basic operation (and compatibility/consistency with WorldEdit) UAK should take:
`[+-]?|x|y|z|h|v`,
`[+-]f[acing|ront] | b[ack|ehind] | l[eft] | r[ight] | u[p] | d[own]`,
`[+-]0-9...`

For Selection Tools UAk should take:
~~`r[ev[erse]]` however `r` by itself should only be recognized as `reverse` if it is at the end of the axis string and not followed by a number (this way we prevent `r[ight]` from conflicting with reverse).~~

`r` should not be a reverse keyword (on second though).
`sym[metrical] | mir[ror] | rev[erse] | true` should be the keywords.

## Value handling:
In the case of sign overlaps (eg. `-x -4`) the signs would be multiplied together (so the result would be the same as `x 4`).

In the case of value overlaps (eg. `x 3 x 6`) values would be summed with all other values *of the same sign* on that axes if UAK is working in 2 vector out mode, or with *all* other values on that axes if UAK is working in single vector out mode.

In cases of orphan values (eg. `x 3 5`) the orphan value (in this case 5) would be added to all other values *of the same sign* on *all* axes if UAK is working in 2 vector out mode, or with *all* other values on *all* axis if UAK is working in single vector out mode.

There may be some cases I'm not remembering but this is it for now.

**Comment by VorTechnix** [comment]
@sbrl What should we do about cases like `front 15 noodle 6`?

**Reply by sbrl** [reply]
That should throw an error due to an unrecognised token `noodle`.
